earth/cpu_intr.c

```diff
@@ -32,13 +31,24 @@ static void timer_reset(uint core_id) {
     mtimecmp_set(mtime_get() + QUANTUM, core_id);
 }
 
+ulonglong gettime() {
+    static ulonglong last_time = 0;
+    ulonglong time = (ulonglong) mtime_get();
+    if (time < last_time) {
+        INFO("gettime() overflows");
+        last_time = time;
+    }
+    return time;
+}
+
 /* Both trap functions are defined in grass/kernel.s */
 void trap_from_M_mode();
 void trap_from_S_mode();
 
 void intr_init(uint core_id) {
     /* Setup the timer */
     earth->timer_reset = timer_reset;
+    earth->gettime = gettime;
```

Comment:
Is `earth->gettime` used anywhere in this PR? If not, I wonder why should we add this function in `struct earth`.